### PR TITLE
Us022 disable respondent enrolment

### PR DESCRIPTION
**Pull request by andrewmil**
Added a new category `DISABLE_RESPONDENT_ENROLMENT` and added some logic to handle this event. This will transition a case to `INACTIONABLE` state and will create a new B-case if there are no remaining actionable cases and the casegroup status is not completed

Works with other PR's:
[response-operations-ui](https://github.com/ONSdigital/response-operations-ui/pull/123)
[rm-backstage](https://github.com/ONSdigital/ras-backstage/pull/96)
[ras-frontstage-api](https://github.com/ONSdigital/ras-frontstage-api/pull/60)
[ras-party](https://github.com/ONSdigital/ras-party/pull/122)
[rm-casesvc-api](https://github.com/ONSdigital/rm-casesvc-api/pull/13)

[Confluence](https://digitaleq.atlassian.net/wiki/spaces/RASB/pages/248545529/US022+Disable+respondent+enrolment+to+a+survey)
[Trello](https://trello.com/c/1ez85L0y/51-us022-int-disable-respondents-enrolment-to-a-survey-l)

### CODE REVIEW

**Comment by gemmai95** [comment]
[US022: INT Disable respondents enrolment to a survey [L]](https://trello.com/c/1ez85L0y/51-us022-int-disable-respondents-enrolment-to-a-survey-l)

**Comment by gemmai95** [comment]
[US022: INT Disable respondents enrolment to a survey [L]](https://trello.com/c/1ez85L0y/51-us022-int-disable-respondents-enrolment-to-a-survey-l)

File: src/test/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImplTest.java [hunk 1]
```diff
@@ -1512,6 +1513,73 @@ public void testEventSuccessfulResponseUploaded() throws Exception {
             any(UUID.class), any(String.class));
   }
 
+  /**
+   * A SUCCESSFUL_RESPONSE_UPLOAD event transitions an actionable BI case to INACTIONABLE, and all associated BI cases in the case group.
```
Comment (tej99) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImplTest.java]: comment needs updating

File: src/test/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImplTest.java [hunk 2]
```diff
@@ -1512,6 +1513,73 @@ public void testEventSuccessfulResponseUploaded() throws Exception {
...
+    verify(caseRepo, times(1)).saveAndFlush(argument.capture());
+    verify(internetAccessCodeSvcClientService, times(1)).disableIAC(any(String.class));
+    verify(caseRepo, times(1)).saveAndFlush(argument.capture());
+    verify(notificationPublisher, times(1)).sendNotification(any(CaseNotification.class));
+    verify(caseRepo, times(1)).findByCaseGroupIdAndState(null, CaseState.ACTIONABLE);
+    verify(caseGroupRepo, times(1)).findById(null);
+  }
+
+  /**
+   * A SUCCESSFUL_RESPONSE_UPLOAD event transitions an actionable BI case to INACTIONABLE, and all associated BI cases in the case group.
```
Comment (tej99) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImplTest.java]: comment needs updating

**Pull request merged**